azkaban-web-server/src/main/java/azkaban/flowtrigger/TriggerInstanceProcessor.java

```diff
@@ -88,22 +91,56 @@ private void executeFlowAndUpdateExecID(final TriggerInstance triggerInst) {
   }
 
   private String generateFailureEmailSubject(final TriggerInstance triggerInstance) {
-    final String flowFullName =
-        triggerInstance.getProjectName() + "." + triggerInstance.getFlowId();
-    return String.format(FAILURE_EMAIL_SUBJECT, flowFullName, this.emailer.getAzkabanName());
+    return String.format(FAILURE_EMAIL_SUBJECT, triggerInstance.getFlowId(), triggerInstance
+        .getProjectName(), this.emailer.getAzkabanName());
   }
 
-  private String generateFailureEmailBody(final TriggerInstance triggerInstance) {
-    final String triggerInstFullName =
-        triggerInstance.getProjectName() + "." + triggerInstance.getFlowId();
-    return String.format(FAILURE_EMAIL_BODY, triggerInstFullName);
+  private EmailMessage createFlowTriggerEmailMessage(final TriggerInstance triggerInst) {
```

Comment:
Rename to `createFailureEmailMessage`? To me it was confusing to find out that this method is actually calling `generateFailureEmailSubject`, so it's not just a "generic notification message" but specifically about a failure case.